### ISSUE TITLE
StackOverflowError in ZMQActor

### ISSUE THREAD

**Issue by btcontract** [ACINQ/eclair#698]
Testnet Eclair GUI node suddenly closed with this output in console: 
```
Uncaught error from thread [eclair-node-gui-akka.actor.default-dispatcher-18]: null, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for for ActorSystem[eclair-node-gui]
java.lang.StackOverflowError
	at zmq.YPipe.read(YPipe.java:132)
	at zmq.Pipe.read(Pipe.java:181)
	at zmq.FQ.recvPipe(FQ.java:76)
	at zmq.FQ.recv(FQ.java:67)
	at zmq.XSub.xrecv(XSub.java:165)
	at zmq.SocketBase.recv(SocketBase.java:662)
	at org.zeromq.ZMQ$Socket.recv(ZMQ.java:1377)
	at org.zeromq.ZFrame.recv(ZFrame.java:337)
	at org.zeromq.ZFrame.recvFrame(ZFrame.java:370)
	at org.zeromq.ZMsg.recvMsg(ZMsg.java:232)
	at fr.acinq.eclair.blockchain.bitcoind.zmq.ZMQActor.checkMsg(ZMQActor.scala:58)
	at fr.acinq.eclair.blockchain.bitcoind.zmq.ZMQActor.checkMsg(ZMQActor.scala:61)
	at fr.acinq.eclair.blockchain.bitcoind.zmq.ZMQActor.checkMsg(ZMQActor.scala:61)
	at fr.acinq.eclair.blockchain.bitcoind.zmq.ZMQActor.checkMsg(ZMQActor.scala:61)
...
```

Software was built from this commit: https://github.com/ACINQ/eclair/commit/12a80afd034751fa4735c188bb6d2cd0155462cb

Can't reproduce this so far.

**Comment by sstone** [comment]
Was your bitcoind syncing when it happened ? There's a potential issue when lots of ZMQ notifications have been received and a proposed fix in #696

**Comment by btcontract** [comment]
> Was your bitcoind syncing when it happened ?

It wasn't, this happened once again just now when bitcoind is fully synchronized.

**Comment by sstone** [comment]
Fixed in #696
